File: Plugins/Aspose_Email_Java_for_Ruby/lib/asposeemailjava/Outlook/addfiletopst.rb
module Asposeemailjava
  module AddFileToPST
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        pst = Rjb::import('com.aspose.email.PersonalStorage').create(data_dir + "AddFile.pst", Rjb::import('com.aspose.email.FileFormatVersion').Unicode)

        fi = pst.createPredefinedFolder("Inbox", Rjb::import('com.aspose.email.StandardIpmFolder').Inbox)

        fi.addFile(data_dir + "Report.xlsx", "IPM.Document.Excel.Sheet.8")

        pst.dispose()

        puts "Added file to PST"
    end
  end
end

File: Plugins/Aspose_Email_Java_for_PHP/src/aspose/email/ProgrammingOutlook/WorkingWithOutlookPersonalStorage/AddFileToPST.php
<?php
namespace Aspose\Email\ProgrammingOutlook\WorkingWithOutlookPersonalStorage;

use com\aspose\email\PersonalStorage as PersonalStorage;
use com\aspose\email\FileFormatVersion as FileFormatVersion;
use com\aspose\email\StandardIpmFolder as StandardIpmFolder;

class AddFileToPST{

    public static function run($dataDir=null){

        $personalStorage=new PersonalStorage();
        $fileFormatVersion=new FileFormatVersion();

        $pst = $personalStorage->create($dataDir . "AddFile2.pst", $fileFormatVersion->Unicode);
        $standardIpmFolder=new StandardIpmFolder();
        $fi = $pst->createPredefinedFolder("Inbox", $standardIpmFolder->Inbox);

        $fi->addFile($dataDir . "Report.xlsx", "IPM.Document.Excel.Sheet.8");

        $pst->dispose();

        print "Added file to PST".PHP_EOL;
    }

}
?>